src/library/scala/collection/IterableOnce.scala

```diff
@@ -43,37 +43,37 @@ final class IterableOnceExtensionMethods[A](private val it: IterableOnce[A]) ext
     case _ => it.iterator.foreach(f)
   }
 
-  @deprecated("Use factory.from(it) instead of it.to(factory) for IterableOnce", "2.13.0")
+  @deprecated("Use it.iterator.to(factory) instead of factory.from(it) for IterableOnce", "2.13.0")
   def to[C1](factory: Factory[A, C1]): C1 = factory.fromSpecific(it)
 
-  @deprecated("Use ArrayBuffer.from(it) instead of it.toBuffer", "2.13.0")
+  @deprecated("Use it.to(ArrayBuffer) instead of ArrayBuffer.from(it)", "2.13.0")
   def toBuffer[B >: A]: mutable.Buffer[B] = mutable.ArrayBuffer.from(it)
 
-  @deprecated("Use ArrayBuffer.from(it).toArray", "2.13.0")
+  @deprecated("Use it.to(ArrayBuffer) instead of ArrayBuffer.from(it)", "2.13.0")
   def toArray[B >: A: ClassTag]: Array[B] = it match {
     case it: Iterable[B] => it.toArray[B]
     case _ => mutable.ArrayBuffer.from(it).toArray
   }
 
-  @deprecated("Use List.from(it) instead of it.toList", "2.13.0")
+  @deprecated("Use it.to(List) instead of List.from(it)", "2.13.0")
```

Comment:
I would drop the `it` reference and directly use `toList`, and we have to go through the `Iterator` also: “Use `.iterator.toList”.